cmd/cli/license/inspect.go

```diff
@@ -0,0 +1,207 @@
+package license
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/spf13/cobra"
+
+	"github.com/bacalhau-project/bacalhau/cmd/util/flags/cliflags"
+	"github.com/bacalhau-project/bacalhau/cmd/util/hook"
+	"github.com/bacalhau-project/bacalhau/cmd/util/output"
+	licensepkg "github.com/bacalhau-project/bacalhau/pkg/lib/license"
+)
+
+type InspectOptions struct {
+	LicenseFile string
+	OutputOpts  output.OutputOptions
+}
+
+func NewInspectOptions() *InspectOptions {
+	return &InspectOptions{
+		OutputOpts: output.OutputOptions{Format: output.TableFormat},
+	}
+}
+
+func NewInspectCmd() *cobra.Command {
+	o := NewInspectOptions()
+	cmd := &cobra.Command{
+		Use:           "inspect",
+		Short:         "Inspect license information",
+		Args:          cobra.NoArgs,
+		PreRun:        hook.ApplyPorcelainLogLevel,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// Check if license file path is empty or just whitespace
+			if o.LicenseFile == "" || len(strings.TrimSpace(o.LicenseFile)) == 0 {
+				return fmt.Errorf("required flag \"license-file\" not set")
+			}
+
+			// Check if license file exists
+			if _, err := os.Stat(o.LicenseFile); os.IsNotExist(err) {
+				return fmt.Errorf("file not found: %s", o.LicenseFile)
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return o.Run(cmd.Context(), cmd)
+		},
+	}
+
+	// Add flags
+	cmd.Flags().StringVar(&o.LicenseFile, "license-file", "", "Path to the license file")
+	cmd.Flags().AddFlagSet(cliflags.OutputFormatFlags(&o.OutputOpts))
+
+	// Set required flag
+	_ = cmd.MarkFlagRequired("license-file")
+
+	return cmd
+}
+
+// LicenseInfo represents the structure of license information
+type LicenseInfo struct {
+	Product        string
+	LicenseID      string
+	CustomerID     string
+	ValidUntil     string
+	LicenseVersion string
+	Capabilities   map[string]string
+}
+
+// jsonOutput is used for JSON/YAML serialization to include metadata
+type jsonOutput struct {
+	LicenseInfo
+	Metadata map[string]string
+}
+
+// MarshalJSON implements custom JSON marshaling
+func (j jsonOutput) MarshalJSON() ([]byte, error) {
+	type Alias jsonOutput // prevent recursion
+	data := struct {
+		Alias
+		Metadata map[string]string `json:"Metadata"`
+	}{
+		Alias:    Alias(j),
+		Metadata: j.Metadata,
+	}
+	return json.Marshal(data)
+}
+
+// MarshalYAML implements custom YAML marshaling
+func (j jsonOutput) MarshalYAML() (interface{}, error) {
+	type Alias jsonOutput // prevent recursion
+	data := struct {
+		Alias
+		Metadata map[string]string `yaml:"Metadata"`
+	}{
+		Alias:    Alias(j),
+		Metadata: j.Metadata,
+	}
+	return data, nil
+}
```

Comment:
are these really needed?

Reply:
@wdbaruni I need that because I am specifically not outputting the metadata in the tabular format.
If we went with other non-tabular output, there will be no need for it.

Reply:
Code removed when the output was changed. 
Resolved